report/3.Theory/SyntaxTheory.tex

```diff
@@ -0,0 +1,178 @@
+If the string does contain right parentheses, which it does. 
+Y cannot be of the length 0.
+
+The first condition is not upheld since if $i = 0$ the new string cannot contain an equal amount of left and right parentheses because of the same reason as to why the second condition is upheld.
+
+This concludes the proof by contradiction.
+Since HCl does not uphold the required property for it to be a regular language, it it not and the only automaton capable of recognizing the language is the NDPA described in the previous sections.
+
+\subsection{Types and Identifiers}
+So far the syntax of HCL has been properly defined as a CFL. 
+Meaning that all Variables, terminals and the transition rules for recognizing HCl has been described in a computational model as seen on page 
+%Indsæt sidetal for HCL CFG definition.
+and in appendix.
+%Indæst reference til appendix hvor NPDA for HCL er beskrevet.
+It is now relevant to define how the types and identifiers in HCL can be described or recognized by a computational model.
+The goal is to formally describe the languages $L_{none}$, $L_{bool}$, $L_{number}$, $L_{text}$ which are the type languages and the identifier language $L_{identifiers}$.
+
+An alphabet is a finite set of symbols, a string is a finite sequence of symbols and a language is a set of strings.
+To describe the languages mentioned above, the Alphabets used in the informal and formal notations needs to be defined. 
+These can be seen in Table 3.1.
+
+\begin{table}[!htb]
+	\centering
+	\begin{tabular}{|l|l|}
+		\hline
+		\textbf{Alphabet} & \textbf{Set}                            \\ \hline
+		T                 & \{true\}                                \\ \hline
+		F                 & \{false\}                               \\ \hline
+		$Z^+$             & The set of all positive integers        \\ \hline
+		$Z^-$             & The set of all negative integers        \\ \hline
+		R                 & The set of all real numbers             \\ \hline
+		D                 & \{$Z^+ \cup Z^-$\}                      \\ \hline
+		$S_p$             & The set of all special characters      \\ \hline
+		$L_l$             & The set of all lower-case letters       \\ \hline
+		$L_u$             & The set of all upper-case letters       \\ \hline
+		C                 & $S_p \cup L_l \cup L_u \cup D$          \\ \hline
+		$R_{eserved}$     & The set of all reserved keywords in HCL \\ \hline
+	\end{tabular}
+	\caption{Table containing all utility alphabets used to formally describe the type and identifier languages}
+\end{table}
+
+By using the set construction method, the languages can now be defined as shown in Table 3.2.
+
+\begin{table}[!htb]
+	\centering
+	\label{my-label}
+	\begin{tabular}{|l|l|}
+		\hline
+		\textbf{Name}     & \textbf{Set}                                    \\ \hline
+		$L_{none}$        & \{w | w $\in$ \{none\}\}                        \\ \hline
```

Comment:
None - not none

Reply:
fixed

Reply:
Igen, tror at han mener at forbogstavet skal stå med stort